siteimprove/admin/class-siteimprove-admin.php

```diff
@@ -164,12 +164,17 @@ public function siteimprove_init() {
 	 */
 	private function siteimprove_add_js( $url, $type ) {
 		$file_name = get_option( 'siteimprove_overlayjs_file', 'overlay-v2-dev.js' );
+		$disabled_new_version = get_option( 'siteimprove_disable_new_version' );
 		$pattern = '/^[a-zA-Z_\d-]+.js/';
 
-		if ( preg_match( $pattern, $file_name ) ) {
+		if ( ! empty( $file_name ) && preg_match( $pattern, $file_name ) ) {
 			$overlay_path = Siteimprove::JS_LIBRARY_URL . $file_name;
 		} else {
-			$overlay_path = $file_name;
+			if ( $disabled_new_version ) {
```

Comment:
Maybe it's just me, but the way I read this, it says "if $disabled_new_version is true then we should use "overlay-latest.js" - and that doesn't seem right to me?

Reply:
(Just tested it and it seems to do the right thing so not sure why I don't understand it. Maybe it's because `$disabled_new_version` is not a bool but a number? In which case the code could be `if ( $disabled_new_version == xx ) { ` to be clearer for someone like me ;-)

Reply:
This variable get an wordpress option, that is stored as 0 (unchecked) or 1 (checked). So, in this case we are verifying if the checkbox is selected or not
![image](https://github.com/Siteimprove/CMS-plugin-Wordpress/assets/106838030/b104f30e-ec6b-4464-a87c-b53cdb28e5d4)